include/swift/RemoteAST/RemoteAST.h

```diff
@@ -211,6 +217,10 @@ class RemoteASTContext {
   /// Given a heap object, resolve its heap metadata.
   Result<remote::RemoteAddress>
   getHeapMetadataForObject(remote::RemoteAddress address);
+
+  /// Project an existential type, given its kind.
+  Result<Type> projectExistentialType(remote::RemoteAddress address,
+                                      remoteAST::ExistentialTypeKind kind);
```

Comment:
What does it mean to project a type?  You mean to read the dynamic type?

Should this return a pair of the type and the address of a value of that type?  Getting the address of the value is not a simple operation on opaque existentials.

Can you just pass in a (static) `Type` that's known to be an existential type instead of hard-coding a set of existential representations into the interface?  Do you not have a `Type` on your side?

Reply:
Thanks for your comments.

* `projectExistential` is the name that the mirrors use for this operation. I guess `readExistentialDynamicType` is probably a better choice.

* Yes, it should, thanks.

* I tried to pass a static type (and in fact, for my use case in the debugger, I can, but I failed to do that when testing using `swift-remoteast-test`. I guess it's just my swift not good enough, can you suggest how to write a test to do that?

Reply:
For what it's worth, when I wrote this I thought I could pass a mangled name to `swift-remoteast-test` and then perform type reconstruction to get the dynamic type. I'm not sure it's really that much better because you need to keep in sync the type and the mangled name, so I decided to not do it.

Reply:
s/to get the dynamic type/to get the static type/